### ISSUE TITLE
Lint/UselessAccessModifier not recognizing delegated methods

### ISSUE THREAD

**Issue by eljojo** [bbatsov/rubocop#2731]
Hi,
The `Lint/UselessAccessModifier` cop doesn't recognize delegated method's (using ActiveSupport).

It considers the following `private` useless.

``` ruby
  private
  delegate :logger, to: :Rails
```

**Comment by alexdowad** [comment]
@bbatsov didn't agree to merge a fix for this issue, as he doesn't want Rails-specific stuff built into RuboCop. Our goal is to get all Rails-related code out of RC core and into some kind of plugin.

**Comment by eljojo** [comment]
I see. Thanks a lot for taking the time to review this :)

**Comment by bbatsov** [comment]
To be more specific - I'm fine with making this particular cop configurable, but I'm not OK with hardcoding Rails specific stuff in generic Ruby cops.

**Comment by eljojo** [comment]
@bbatsov should I re-open this?

**Comment by bbatsov** [comment]
I've reopened it. If you're up for a bit of hacking you can make the cop configurable yourself. It's pretty easy, but unfortunately I'm extremely short on time.

**Comment by Drenmi** [comment]
I've been thinking a bit about this one, and can off the top of my head see two routes we could take.

1. Add a `SafeMode` which accepts it as long as any DSL/class method is called in the scope of the modifier. (Simpler for the user, but less flexible, and more prone to false negatives.)

2. Add a configurable list of acceptable DSL/class methods. (More complex for user, but allows to fully express intent.)

I'm leaning towards option 2. What do you think @alexdowad, @bbatsov?

**Comment by alexdowad** [comment]
Option 2 is better.
